kevm-pyk/src/kevm_pyk/kproj/evm-semantics/lemmas/slot-updates.k

```diff
@@ -0,0 +1,144 @@
+requires "word.md"
+
+module SLOT-UPDATES [symbolic]
+    imports EVM-TYPES
+    imports BUF
+    imports WORD
+
+    // ###########################################################################
+    // Auxiliaries
+    // ###########################################################################
+
+    // Index of first bit that equals one
+    syntax Int ::=  #getFirstOneBit(Int) [function, total]
+
+    rule [gfo-succ]: #getFirstOneBit(X:Int) => log2Int ( X &Int ( ( maxUInt256 xorInt X ) +Int 1 ) ) requires          #rangeUInt(256, X) andBool X =/=Int 0
+    rule [gfo-fail]: #getFirstOneBit(X:Int) => -1                                                    requires notBool (#rangeUInt(256, X) andBool X =/=Int 0)
```

Comment:
Huh, this is pretty clever non-looping algorithm. I guess we could also repeatedly divide by 2, but that could be slower.

I wonder though, should we mark this rule (or this fnuction) as `concrete`, so that it doesn't try to build this RHS expression for `X` when it's symbolic?

Reply:
I did have that, but then it was not passed to the LLVM backend correctly and I was getting undefined evaluations. @jberthold advised to get rid of all attributes and then it worked. It is called only with concrete values in the associated simplifications below.